### PR TITLE
Abstract store

### PR DESCRIPTION
**Pull request by nocturnalastro**

### CODE REVIEW

File: skeleton_etl/stores/abstract.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @abstractmethod
+    def __iter__(self):
+        pass
+
+    @abstractmethod
+    def load(self, **kwargs):
+        pass
+
+    @abstractmethod
+    def save(self, **kwargs):
```
Comment (crwr45) [inline skeleton_etl/stores/abstract.py:32]: Between the abstract and the example implementation it's not very clear what the `save` method is meant to do in the general case. Please can you document return values and intended purpose to address that?

File: skeleton_etl/stores/memory.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+class InMemoryStore(AbstractStore):
+    def __init__(self):
+        self._store = {}
+
+    def __len__(self):
+        return len(self._store)
+
+    def __iter__(self):
+        for _, value in sorted(self._store.items()):
```
Comment (crwr45) [inline skeleton_etl/stores/memory.py:12]: Any reason not to use `self._store.values()`?

**Review by crwr45 — CHANGES_REQUESTED** [review]
A sensible addition, minor improvements needed.

File: skeleton_etl/stores/abstract.py (hunk 1 above)
Comment (nocturnalastro) [reply skeleton_etl/stores/abstract.py:32]: Its meant to allow you to persist the data but doing that with an in-memory one requires a lot of assumptions about storage medium (pickle, numpy file, json dump, etc...) 

suppose you could allow them to pass an object in the in-memory one to persist the data but I thought as a start we could just return the list.

File: skeleton_etl/stores/memory.py (hunk 2 above)
Comment (nocturnalastro) [reply skeleton_etl/stores/memory.py:12]: I was tired ... no idea, good spot.

Comment (nocturnalastro) [reply skeleton_etl/stores/memory.py:12]: Oh, wait I remember now it is returned in key order.

Comment (crwr45) [reply skeleton_etl/stores/memory.py:12]: Good reason

File: skeleton_etl/stores/abstract.py (hunk 1 above)
Comment (crwr45) [reply skeleton_etl/stores/abstract.py:32]: I agree there's probably two concepts here: `dump` to give a standardised representation of the data in the store, and `save` to persist the data in the method appropriate for the store.

In the case of the `InMemoryStore`, a call to `save` is probably a no-op, while `dump` returns the list.

**Pull request merged**